### PR TITLE
[core] Add a req file to list all the Python checks shipped with the agent

### PR DESCRIPTION
**Pull request by masci**
### What does this PR do?

Adds a requirement file detailing which checks are included in a certain version of the agent. At the moment this file is not used anywhere in the build process, mid-term goal will be using `pip` to actually install and include in the package the desired checks at Agent build time.

### Motivation

At the moment, determining which checks were shipped with which Agent is a complex and esoteric procedure. We need a single source of truth to determine the list of packages built with the agent.

### CODE REVIEW

File: requirements-integrations-core.txt [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+teamcity==1.0.0
+tokumx==1.2.0
+twemproxy==1.2.0
+varnish==1.2.0
+vsphere==2.1.0
+win32_event_log==1.1.1; sys_platform == 'windows'
+windows_service==1.2.0; sys_platform == 'windows'
+wmi_check==1.1.1; sys_platform == 'windows'
+yarn==1.2.0
+zk==1.2.0; sys_platform != 'windows'
```
Comment (ofek) [inline requirements-integrations-core.txt]: `zk` check has no deps

**Review by ofek — COMMENTED** [review]
Great idea! Just a few remarks:

**Review by ofek — CHANGES_REQUESTED** [review]
Also https://github.com/DataDog/integrations-core/pull/1557#discussion_r188690143

File: requirements-integrations-core.txt [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+hdfs_datanode==1.2.0; sys_platform != 'win32'
+hdfs_namenode==1.2.1; sys_platform != 'win32'
+http_check==2.1.0
+iis==2.1.0; sys_platform == 'win32'
+istio==1.0.0
+kafka_consumer==1.4.0; sys_platform != 'win32'
+kong==1.2.0
+kube_dns==1.3.0
+kube_proxy==1.0.0
+kubelet==1.2.0; sys_platform != 'win32'
```
Comment (olivielpeau) [inline requirements-integrations-core.txt]: should be `linux` only (i.e. mac should be excluded too)

**Review by ofek — DISMISSED** [review]

**Review by olivielpeau — COMMENTED** [review]
one nit (there might be other small platform exclusion mistakes I've missed though)

**Review by ofek — APPROVED** [review]

**Comment by masci** [comment]
Closing, we eventually hosted the file in `integrations-core`

**Pull request closed without merging**